src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] arrayList;
+    private int size;
+
+    ArrayList() {
+        arrayList = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        grow();
+        if (index != size) {
+            indexCheck(index);
+            System.arraycopy(arrayList, index, arrayList, index + 1, size - index);
+        }
+        arrayList[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexCheck(index);
+        return arrayList[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexCheck(index);
+        arrayList[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexCheck(index);
+        T element = arrayList[index];
+        size--;
+        System.arraycopy(arrayList, index + 1, arrayList, index, size - index);
+        return element;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (element == arrayList[i] || element != null && element.equals(arrayList[i])) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException("Can't find element " + element);
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    @Override
+    public int indexOf(T element) {
+        return indexOfRange(element, size);
+    }
+
+    private int indexOfRange(T element, int end) {
+        if (element == null) {
+            for (int i = 0; i < end; i++) {
+                if (arrayList[i] == null) {
+                    return i;
+                }
+            }
+        } else {
+            for (int i = 0; i < end; i++) {
+                if (element.equals(arrayList[i])) {
+                    return i;
+                }
+            }
+        }
+        return -1;
+    }
```

Comment:
you don't need to put `size` as parameter.
```suggestion
    private int indexOf(T element) {
        if (element == null) {
            for (int i = 0; i < size; i++) {
                if (arrayList[i] == null) {
                    return i;
                }
            }
        } else {
            for (int i = 0; i < size; i++) {
                if (element.equals(arrayList[i])) {
                    return i;
                }
            }
        }
        return -1;
    }
```